packages/dd-trace/src/scope/base.js

```diff
@@ -68,30 +68,33 @@ class Scope {
 
   _bindEmitter (emitter, span) {
     if (!this._isEmitter(emitter)) return emitter
-
-    wrapMethod(emitter, 'addListener', wrapAddListener, this, span)
-    wrapMethod(emitter, 'prependListener', wrapAddListener, this, span)
-    wrapMethod(emitter, 'on', wrapAddListener, this, span)
-    wrapMethod(emitter, 'once', wrapAddListener, this, span)
-    wrapMethod(emitter, 'prependOnceListener', wrapAddListener, this, span)
-
-    wrapMethod(emitter, 'removeListener', wrapRemoveListener)
-    wrapMethod(emitter, 'off', wrapRemoveListener)
-
-    wrapMethod(emitter, 'removeAllListeners', wrapRemoveAllListeners)
-
+    if (!emitter.__is_dd_emitter) {
+      Scope._wrapEmitter(emitter)
+    }
+    emitter.__dd_span = span
+    emitter.__dd_scope = this
     return emitter
   }
 
+  // Occaisionally we want to wrap a prototype rather than emitter instances,
+  // so we're exposing this as a static method. This lets gives us a faster
+  // path for binding instances of known EventEmitter subclasses.
+  static _wrapEmitter (emitter) {
+    wrapAddListener('addListener', emitter)
+    wrapAddListener('prependListener', emitter)
+    wrapAddListener('on', emitter)
+    wrapAddListener('once', emitter)
+    wrapAddListener('prependListener', emitter)
+    wrapRemoveListener('removeListener', emitter)
+    wrapRemoveListener('off', emitter)
+    wrapRemoveAllListeners(emitter)
+    emitter.__is_dd_emitter = true
+  }
+
   _unbindEmitter (emitter) {
     if (!this._isEmitter(emitter)) return emitter
-
-    unwrapMethod(emitter, 'addListener')
```

Comment:
Should there be a way to unwrap the prototype?

Reply:
If emitters aren't bound, then the prototype methods function exactly as they would if they were unwrapped, so I don't see much use in enabling this functionality (which would also complicate things by needing to store the unwrapped methods).

Reply:
It's more of a safety net if there is any bug or big regression, but since right now the base `EventEmitter` class is not patched, it should have minimum impact so I guess it's fine to not have it.